### PR TITLE
Simplify some tests.

### PR DESCRIPTION
**Pull request by pkofod**
The idea is to remove a lot of the direct use of `*Differentiable` as we are moving towards discouraging this format for most users.

### CODE REVIEW

File: test/newton.jl [hunk 1]
```diff
@@ -70,9 +68,9 @@
         		ddf = OnceDifferentiable(prob.f, prob.g!)
```
Comment (anriseth) [inline test/newton.jl]: Do you need `ddf` now? Or should we just pass `prob.f, prob.g!, prob.h!` on line 69?

Comment (pkofod) [inline test/newton.jl]: Good question. I think I left it in just to check that all "valid" input variations would work. Maybe there should be more comments in the tests.

Comment (anriseth) [inline test/newton.jl]: Or we could have a separate testset for tests where we pass in *Differentiable types?

Comment (pkofod) [inline test/newton.jl]: That would work. I'll update this or to reflect that

Comment (pkofod) [inline test/newton.jl]: Or = pr (can't edit on phone)

File: test/newton.jl [hunk 2]
```diff
@@ -50,29 +47,25 @@
...
+        res = optimize(prob.f, prob.g!, prob.h!, [0., 0.], Newton())
         @test norm(Optim.minimizer(res) - prob.solutions) < 1e-9
     end
 
     @testset "Optim problems (ForwardDiff)" begin
         for (name, prob) in Optim.UnconstrainedProblems.examples
         	if prob.istwicedifferentiable
-        		ddf = OnceDifferentiable(prob.f, prob.g!)
-        		res = Optim.optimize(ddf, prob.initial_x, Newton(), Optim.Options(autodiff = true))
         		@test norm(Optim.minimizer(res) - prob.solutions) < 1e-2
```
Comment (anriseth) [inline test/newton.jl]: This line no longer makes sense (res is not defined yet)

Comment (pkofod) [inline test/newton.jl]: I know, its filing locally ;)

**Comment by pkofod** [comment]
I wonder if something breaks stuff on v0.6. Check this out
```

**Comment by anriseth** [comment]
The v0.4 results are the same as v0.6.

**Comment by pkofod** [comment]
> The v0.4 results are the same as v0.6.
> 
> EDIT: And the difference between v0.6 and v0.5 only happens for finite differencing, not for autodiff.

Strange, exactly the same packages are installed (same versions) no matter the JuliaLang vers. I guess I need to install v0.6 to inspect.

**Comment by pkofod** [comment]
> EDIT: And the difference between v0.6 and v0.5 only happens for finite differencing, not for autodiff.

It's pretty weird. I get differences in results, but I don't get different gradients if I check them out manually at various `x`s across julia versions. More prints!

**Comment by pkofod** [comment]
Oh my... I think I finally found out why the tests might be failing sometimes. It's the initialization of `y` in accelerated gradient descent that sometimes would contain a nan, such that `y_previous` would contain a nan, such that even if `y` is overwritten, `x` would still be corrupted in the end. I think that particular solver needs to be bug checked thoroughly, because it has had some weird performance (optimization wise) behavior in the past as well.

**Pull request merged**